app/javascript/packages/password-confirmation/password-confirmation-element.spec.ts

```diff
@@ -0,0 +1,99 @@
+import userEvent from '@testing-library/user-event';
+import { fireEvent, getByLabelText, waitFor } from '@testing-library/dom';
+import { useSandbox } from '@18f/identity-test-helpers';
+import * as analytics from '@18f/identity-analytics';
+import './password-confirmation-element';
+import type PasswordConfirmationElement from './password-confirmation-element';
+
+describe('PasswordConfirmationElement', () => {
+  let element: PasswordConfirmationElement;
+  let input1: HTMLInputElement;
+  let input2: HTMLInputElement;
+  let idCounter = 0;
+  const sandbox = useSandbox();
+
+  function createElement() {
+    element = document.createElement('lg-password-confirmation') as PasswordConfirmationElement;
+    const idSuffix = ++idCounter;
+    element.innerHTML = `
+      <label for="input-${idSuffix}">Password</label>
+      <input id="input-${idSuffix}" class="password-confirmation__input">
+      <label for="input-${idSuffix}b">Confirm password</label>
+      <input id="input-${idSuffix}b" class="password-confirmation__input-confirmation">
+      <div class="password-confirmation__toggle-wrapper">
+        <input
+          id="toggle-${idSuffix}"
+          type="checkbox"
+          class="password-confirmation__toggle"
+          aria-controls="input-${idSuffix}"
+        >
+        <label for="toggle-${idSuffix}" class="usa-checkbox__label password-confirmation__toggle-label">
+          Show password
+        </label>
+      </div>`;
+    document.body.appendChild(element);
+    return element;
+  }
+
+  beforeEach(() => {
+    element = createElement();
+    input1 = getByLabelText(element, 'Password') as HTMLInputElement;
+    input2 = getByLabelText(element, 'Confirm password') as HTMLInputElement;
+  });
+
+  it('initializes input type', () => {
+    expect(input1.type).to.equal('password');
+  });
+
+  it('changes input type on toggle', async () => {
+    const toggle = getByLabelText(element, 'Show password') as HTMLInputElement;
+
+    await userEvent.click(toggle);
+
+    expect(input1.type).to.equal('text');
+  });
+
+  it('logs an event when clicking the Show Password button', async () => {
+    sandbox.stub(analytics, 'trackEvent');
+    const toggle = getByLabelText(element, 'Show password') as HTMLInputElement;
+
+    await userEvent.click(toggle);
+
+    expect(analytics.trackEvent).to.have.been.calledWith('Show Password button clicked', {
+      path: window.location.pathname,
+    });
+  });
+
+  describe('Password validation', () => {
+    it('validates passwords in both directions', async () => {
+      await userEvent.type(input1, 'salty pickles');
+      fireEvent.input(input1);
```

Comment:
I wouldn't expect you need to `fireEvent` in this spec, since `userEvent` should be handling all of the events associated with a real user interaction. In fact, I'd be concerned if we did need it, since it would imply that the behaviors we're expecting don't work based on the user interaction alone. But when I tested, the spec passed when removing all of the `fireEvent`, so I think we can safely drop them.

Reply:
Went from a spartan style to overly verbose. I figured it wouldn't hurt. But will remove.